### ISSUE TITLE
java.util.ConcurrentModificationException in download process

### ISSUE THREAD

**Issue by tobsecret** [nextflow-io/nextflow#971]
### Expected behavior and actual behavior

My workflow currently exists only of a process to download samples (as well as a process to download and index the reference but that one seems to be unconnected to the problem) and is expected to simply download files. Instead it yields me a concurrentModificationException after downloading 500 or so files and then consistently yields that error every time I use the `-resume` flag. 
Unfortunately it requires software that is not conda installable:
https://github.com/enasequence/enaBrowserTools

```
/*
This is a workflow for my aim1. As such it will include downloading P. falciparum and P. vivax genomic data, aligning it to the respective reference genomes, calling SNPs, filtering SNPs, producing mutational signatures from filtered SNPs, etc.
*/
species = [
           'Plasmodium-falciparum': [
                                     'fastaURL':'http://plasmodb.org/common/downloads/Current_Release/Pfalciparum3D7/fasta/data/PlasmoDB-41_Pfalciparum3D7_Genome.fasta',
                                     'gffURL':'http://PlasmoDB.org/common/downloads/Current_Release/Pfalciparum3D7/gff/data/PlasmoDB-41_Pfalciparum3D7.gff',
                                     ],
/*           'Plasmodium-vivax': [
                                'fastaURL':'http://PlasmoDB.org/common/downloads/Current_Release/PvivaxSal1/fasta/data/PlasmoDB-37_PvivaxSal1_Genome.fasta',
                                'gffURL':'http://PlasmoDB.org/common/downloads/Current_Release/PvivaxSal1/gff/data/PlasmoDB-37_PvivaxSal1.gff'
                                ], */
          ]

params.AccessionsFile = "${baseDir}/Pfalciparum_samples_85percentcallable_Panoptes.txt"
falciparum_accessions = Channel
                  .fromPath(params.AccessionsFile)
                  .splitCsv(header: true, sep: '\t', strip: true)
                  .map {row -> row.ena_accession}
                  .splitCsv(header: false, sep: ',')
                  .flatten()
                  //.take(2)
                  //.subscribe{it -> println it}


process download_falciparum_reads {
    /*
    This process downloads fastq.gz files from ENA accessions. I run enaDataGet for accessions which point to
    only one sequencing lane, i.e. one pair of fastq files. It fails for accessions that point to multiple
    lanes, so in that case I use enaGroupGet instead. Both of these utilities are supplied by ENA:
    https://github.com/enasequence/enaBrowserTools
    */
    tag "$accession"
    publishDir 'reads', mode: 'symlink'
    //storeDir '/scratch/schrat01-share/aim1_mutational_signatures_dna_repair/data/falciparum/reads'
    cpus 1
    time '1h'
    errorStrategy 'retry'
    maxRetries 3
    memory {8.GB * task.attempt}
    time {1.hour * task.attempt}
    maxForks 50 //--> means only 50 of this process can run at a time. This is necessary because the cluster only has so much download bandwidth and the download script fails disproportionally often if too many processes are running at a time.

    input:
    val accession from falciparum_accessions

    output:
    set accession, file('**.fastq.gz') into downloaded_falciparum_reads //possibly should be '**.fastq*' to include both .fastq.gz and .fastq files.

    script:
    """
    enaGroupGet -f fastq $accession
    number_of_files=\$(find . -name '*.fastq.gz' | wc -l)
    exit \$((\$number_of_files%2)) #checks if there are an even number of files --> exits with 0 if yes, and 1 if not
    """

}

paired_falciparum_reads = downloaded_falciparum_reads
                                                     .map{ it-> [it[0], it[1].sort().collate(2)] }
                                                     .transpose()
                                                     .map{ [it[0], file(it[1][0]), file(it[1][1])] }
/*
process download_reference {
    /*
    This process downloads the reference genomes specified in the dictionary (called a map in groovy) species. I am downloading these from PlasmoDB, who unfortunately do not provide permanent links, so expect this to fail.
    */
    tag "$species"
    publishDir 'reference', mode: 'symlink'
    cpus 1
    time '1h'
    memory '20GB'

    input:
    val species from species.collect {it.key}
    val referenceFastaURL from species.collect {it.value.fastaURL}
    val referenceGffURL from species.collect {it.value.gffURL}

    output:
    set species, file('*.fa*') into referenceFasta
    file '*.gff*' into referenceGff

    script:
    """
    wget $referenceFastaURL
    wget $referenceGffURL
    """
}

process index_reference {
    /*
    This process indexes the reference using bwa index
    */
    tag "$species"
    publishDir 'reference', mode: 'symlink'
    cpus 1
    time '1h'
    memory '30GB'
    conda 'bioconda::bwa=0.7.17 bioconda::samtools=1.9 bioconda::picard=2.18.7 bioconda::gatk4=4.0.5.1'

    input:
    set val(species), file(referenceFasta) from referenceFasta.first()

    output:
    set file(referenceFasta), file("${referenceFasta.baseName}.*") into indexed_references

    script:
    """
    bwa index $referenceFasta
    samtools faidx $referenceFasta
    picard CreateSequenceDictionary REFERENCE=$referenceFasta
    """
}
indexed_references = indexed_references.collect()

```

### Steps to reproduce the problem
This is the input file needed for the workflow.
[Pfalciparum_samples_85percentcallable_Panoptes.txt](https://github.com/nextflow-io/nextflow/files/2677751/Pfalciparum_samples_85percentcallable_Panoptes.txt)

### Program output 

[85/8c7bbb] Submitted process > download_falciparum_reads (ERS071779)
ERROR ~ Error executing process > 'download_falciparum_reads (ERS025287)'

Caused by:
  java.util.ConcurrentModificationException

 -- Check '.nextflow.log' file for details
WARN: Killing pending tasks (66)

[.nextflow.log](https://github.com/nextflow-io/nextflow/files/2677742/default.nextflow.log)

### Environment 

* Nextflow version:  18.10.1 
* Java version: openjdk version "1.8.0_131"

* Operating system: Linux `Linux bigpurple-ln4 3.10.0-693.17.1.el7.x86_64 #1 SMP Sun Jan 14 10:36:03 EST 2018 x86_64 x86_64 x86_64 GNU/Linux`

### Additional context

As one can see in the `nextflow.log`, processes are often enough not counted as properly submitted:
```
[ff/21f8e3] NOTE: Error submitting process 'download_falciparum_reads (ERS360542)' for execution -- Execution is retried (1)
WARN: [SLURM] queue status cannot be fetched > exit status: 1
```
These processes tend to still run, so it might be that this is giving NextFlow troubles, as per my retry policy, the "failed to submit" processes get resubmitted, so they effectively run more than once. I am however not sure if this is the cause of the concurrentModificationException.

**Comment by tobsecret** [comment]
Not sure the patch installed correctly on my system :thinking: 
```
[schrat01@bigpurple-ln4 aim1]$ NXF_VER=18.12.0-SNAPSHOT NXF_OPTS='-Xms512m -Xmx2G' nextflow run -resume ~/projects/aim1_pfalciparum_mutational_signatures_and_dna_repair/pipeline/pipeline_pfalciparum.nf
CAPSULE: Downloading dependency io.nextflow:nxf-httpfs:pom:18.12.0-20181213.222210-8
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.nextflow:nxf-httpfs:pom:18.12.0-20181213.222210-8 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency io.nextflow:nxf-httpfs:pom:18.12.0-20181213.222210-8
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-json:pom:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-json:pom:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verb
ose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:pom:2.5.4
CAPSULE: Downloading dependency io.nextflow:nxf-commons:pom:18.12.0-20181213.221705-8
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.nextflow:nxf-commons:pom:18.12.0-20181213.221705-8 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency io.nextflow:nxf-commons:pom:18.12.0-20181213.221705-8
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-templates:pom:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-templates:pom:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log
=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:pom:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-nio:pom:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-nio:pom:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verbo
se)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:pom:2.5.4
CAPSULE: Downloading dependency io.nextflow:nextflow:pom:18.12.0-20181213.221528-6
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact io.nextflow:nextflow:pom:18.12.0-20181213.221528-6 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency io.nextflow:nextflow:pom:18.12.0-20181213.221528-6
CAPSULE: Downloading dependency org.codehaus.groovy:groovy:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy:pom:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy:pom:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy:pom:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-xml:pom:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:pom:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-xml:pom:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verbo
se)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:pom:2.5.4
CAPSULE: Downloading dependency io.nextflow:nxf-httpfs:jar:18.12.0-20181213.222210-8
CAPSULE: Downloading dependency io.nextflow:nxf-commons:jar:18.12.0-20181213.221705-8
CAPSULE: Downloading dependency io.nextflow:nextflow:jar:18.12.0-20181213.221528-6
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:jar:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-json:jar:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-templates:jar:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-nio:jar:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy:jar:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-xml:jar:2.5.4 in local (file:/gpfs/home/schrat01/.m2/repository) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:jar:2.5.4
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy:jar:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verbose)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-xml:jar:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verbo
se)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-json:jar:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verb
ose)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-nio:jar:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log=verbo
se)
CAPSULE: Transfer failed: capsule.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.codehaus.groovy:groovy-templates:jar:2.5.4 in https://oss.sonatype.org/content/repositories/snapshots (https://oss.sonatype.org/content/repositories/snapshots) (for stack trace, run with -Dcapsule.log
=verbose)
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy:jar:2.5.4
CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:jar:2.5.4
N E X T F L O W  ~  version 18.12.0-SNAPSHOT
Launching `/gpfs/home/schrat01/projects/aim1_pfalciparum_mutational_signatures_and_dna_repair/pipeline/pipeline_pfalciparum.nf` [evil_nobel] - revision: 97a7aa3122
WARN: It appears you have never run this project before -- Option `-resume` is ignored
WARN: There's no process matching config selector: align_reads_bwa
[warm up] executor > slurm
```
In one instance the executor failed to fetch the queue status:
```
[63/f54224] NOTE: Error submitting process 'download_falciparum_reads (SRS312023)' for execution -- Execution is retried (1)
WARN: [SLURM] queue status cannot be fetched
- cmd executed: squeue --noheader -o %i %t -t all -u schrat01
- exit status : 1
- output      :
  slurm_load_jobs error: Socket timed out on send/recv operation
[47/544f89] NOTE: Error submitting process 'download_falciparum_reads (ERS193640)' for execution -- Execution is retried (1)
```
And ended again around 500-ish downloaded samples with a concurrentModificationException:
```
[3f/e64d36] NOTE: Process `download_falciparum_reads (SRS312024)` terminated with an error exit status (143) -- Execution is retried (1)
ERROR ~ Error executing process > 'download_falciparum_reads (ERS188089)'

Caused by:
  java.util.ConcurrentModificationException


 -- Check '.nextflow.log' file for details
[34/09d42e] Re-submitted process > download_falciparum_reads (SRS312024)
WARN: Killing pending tasks (58)
[a2/715065] Submitted process > download_falciparum_reads (ERS199601)
```

Attached the log-file:
[.nextflow_18.12.0-SNAPSHOT.log](https://github.com/nextflow-io/nextflow/files/2681405/default.nextflow_18.12.0-SNAPSHOT.log)

**Comment by pditommaso** [comment]
The messages `CAPSULE: Transfer failed .. etc` are expected because it's a snapshot version, however still the main problem is not solved.  

Also I'm start to think `ConcurrentModificationException` is a side effect of the `slurm_load_jobs error: Socket timed out on send/recv operation`. That signals that your cluster cannot cope with the submission rate of NF. You can try to set a `executor. submitRateLimit = '10/s'` to slow it down. 

See https://www.nextflow.io/docs/latest/config.html#scope-executor

**Comment by pditommaso** [comment]
I withdraw what I wrote the slurm error look unrelated from the ConcurrentModificationException, still investigating.

**Comment by tobsecret** [comment]
I may have found the problem: Actual duplicated accessions in the input file... I am so sorry - that's such a stupid mistake on my part. Should have sanity-checked the file before I used it.

**Comment by pditommaso** [comment]
I can't see that, where is it?

**Comment by tobsecret** [comment]
The following one-liner shows that there are duplicate records in Pfalciparum_samples_85percentcallable_Panoptes.txt:
`cut -f5 Pfalciparum_samples_85percentcallable_Panoptes.txt | sort | uniq -c | tr -s " " | cut -f2 -d' ' | sort | uniq -c`
```
 1732 1
  102 2
```

**Comment by pditommaso** [comment]
Not sure that it can cause the `ConcurrentModificationException`. That happen because for some reason the list of output files change while the process starts to publish them.

**Comment by pditommaso** [comment]
Error trace 

```
java.util.ConcurrentModificationException: null
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
	at java.util.ArrayList$Itr.next(ArrayList.java:851)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2295)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2281)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2322)
	at org.codehaus.groovy.runtime.dgm$186.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:246)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:55)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at nextflow.processor.TaskProcessor$_publishOutputs0_closure22.doCall(TaskProcessor.groovy:1198)
	at sun.reflect.GeneratedMethodAccessor1173.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:104)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:326)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041)
	at groovy.lang.Closure.call(Closure.java:411)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.callClosureForMapEntry(DefaultGroovyMethods.java:5927)
	at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2373)
	at org.codehaus.groovy.runtime.dgm$187.invoke(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:246)
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:55)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at nextflow.processor.TaskProcessor.publishOutputs0(TaskProcessor.groovy:1193)
	at nextflow.processor.TaskProcessor.publishOutputs(TaskProcessor.groovy:1182)
```

**Comment by tobsecret** [comment]
The changes in `submitRateLimit` and other executor modifications did not change the behaviour. 
Will try and see if deduplicating my input file solves the problem.

**Comment by pditommaso** [comment]
Bad. However if you are referring the error 

```
WARN: [SLURM] queue status cannot be fetched
- cmd executed: squeue --noheader -o %i %t -t all -u schrat01
- exit status : 1
- output      :
  slurm_load_jobs error: Socket timed out on send/recv operation
```

this is supposed to be a different issue. could you please open a separate issue for that. 

For the `ConcurrentModificationException` I've merged a possible patch. You can try it using exporting the variable `NXF_VER=18.12.0-edge`, running NF as usual.

**Comment by tobsecret** [comment]
Hey @pditommaso, sorry - the last few days have been reasonably busy. 
I am running the new `NXF_VER=18.12.0-edge` right now - will let you know how it goes!

Yes, we are using the same cluster! We have been having issues lately because of the cluster being modified, so #970 could be related to our cluster.

**Comment by tobsecret** [comment]
Looking good so far! Has downloaded 1670 samples when it would usually die at around 500 samples.

**Comment by pditommaso** [comment]
🤞

**Comment by pditommaso** [comment]
Any good news here?

**Comment by pditommaso** [comment]
Excellent.! Happy holidays!

 Solved by [27d9eb65]
